src/lib389/lib389/utils.py

```diff
@@ -1701,10 +1708,66 @@ def is_valid_hostname(hostname):
     return all(allowed.match(x) for x in hostname.split("."))
 
 
+def parse_size(size):
+    """
+    Parse a string representing a size (like "5 kb" or "2.5Gb") and
+    return the size in bytes.
+    :param size: The size to parse
+    :type size: str:
+    :return int:
+    :raise ValueError: if the string cannot be parsed.
+    """
+    try:
+        val,unit = re.fullmatch(SIZE_PATTERN, size, flags=re.IGNORECASE).groups()
+        return round(float(val) * SIZE_UNITS[unit.lower()])
+    except AttributeError:
+        raise ValueError(f'Unable to parse "{size}" as a size.')
+
+
+def format_size(size):
+    """
+    Return a string representing a size (like "5 kb" or "2.5Gb")
+    :param size: The size int bytes to format
+    :type size: int:
+    :return str:
+    """
+    for unit in ["B", "KB", "MB", "GB",]:
+        if (size < 1024):
+            return f"{size:3.1f} {unit}"
+        size /= 1024
+    return f"{size:.1f} TB"
+
+
 def get_default_db_lib():
+    """
+    Get the default value for the database implementation
+    :return str:  Should be either 'bdb' or 'mdb'
+    """
     return os.getenv('NSSLAPD_DB_LIB', default=DEFAULT_DB_LIB)
 
 
+def get_default_mdb_max_size(paths):
+    """
+    Get the default maximum size for the lmdb database.
+    :return str:  A size that can be parsed with parse_size()
+    """
+    if paths is None:
+        paths = Paths()
+    mdb_max_size = DEFAULT_LMDB_SIZE
+    size = parse_size(mdb_max_size)
+    # Make sure that there is enough available disk space
+    # otherwise decrease the value
+    try:
+        statvfs = os.statvfs(paths.db_dir)
+        avail = statvfs.f_frsize * statvfs.f_bavail
+        avail *= 0.8 # Reserve 20% as margin
+        if size > avail:
+            mdb_max_size = str(avail)
+    except Exception as e:
+        pass
```

Comment:
I think we should at least log the error to `debug` log level... But ideally, we would skip only the expected error types (not all of  them).

Reply:
I agree: it should have been OSError instead of Exception ...
If we choose to log a message, we also need to change the test. 
 (Because FileNotFound is expected when using the default Paths 
    (instance is not known and the path is  /var/lib/dirsrv/slapd-{instance_name}/db  ) 
I guess we have better retry with the parent path.
Working on it ...

Reply:
It is fixed. 
Also added a small change to display properly the DSLdapObject subclass objects:
Will now display:
`E       AssertionError:  assert not [<class 'lib389.idm.account.Account'>(instance_name="standalone1", dn="uid=demo_user,ou=people,dc=example,dc=com"), <class 'lib389.idm.account.Account'>(instance_name="standalone1", dn="cn=demo_group,ou=Groups,dc=example,dc=com")]
`
Instead of:
`E       AssertionError: assert not [<lib389.idm.account.Account object at 0x7fcc849ea790>, <lib389.idm.account.Account object at 0x7fcc849eb150>]`

Reply:
Looks good! Thanks!